lib/core/src/Cardano/Pool/Performance.hs

```diff
@@ -0,0 +1,155 @@
+  where
+    b = fromIntegral $ natVal $ Proxy @b
+
+data RankingConstants = RankingConstants
+    { leaderStakeInfluence :: Double >= 0
+      -- ^ a_0
+    , punishmentHarshness :: Double >= 0
+      -- ^ \gamma
+    , optimalNumberOfPools :: Int >= 1
+      -- ^ k
+    , totalRewards :: Lovelace >= 1
+      -- ^ Total rewards in an epoch. "R" in the spec.
+    } deriving (Show, Eq, Generic)
+
+saturatedPoolSize :: RankingConstants -> RelativeStakeOf "pool"
+saturatedPoolSize env =
+    RelativeStake $ 1 / fromIntegral (unBounded $ optimalNumberOfPools env)
+
+data Pool = Pool
+    { leaderStake :: RelativeStakeOf "pool leader"
+    , cost :: Lovelace
+    , margin :: Margin
+    , recentAvgPerformance :: Double >= 0
+      -- ^ An already averaged performance-value
+    } deriving (Show, Eq, Generic)
+
+newtype Lovelace = Lovelace { getLovelace :: Double }
+    deriving (Show, Eq)
+    deriving newtype (Ord, Num)
+newtype Margin = Margin { unMargin :: Double }
+    deriving (Show, Eq)
+    deriving newtype Ord
+
+-- The value
+-- 0.01 :: RelativeStakeOf "pool"
+-- would mean that a pool has a stake that is 1% of the total active stake in
+-- the epoch.
+newtype RelativeStakeOf (tag :: Symbol)
+    = RelativeStake { unRelativeStake :: Double }
+    deriving (Eq, Ord, Show, Generic)
+    deriving newtype (Num, Fractional)
+
+--------------------------------------------------------------------------------
+-- Formulas from spec
+--------------------------------------------------------------------------------
+
+--
+-- Non-myopic utility
+--
+-- "From Design Specification for Delegation and Incentives in Cardano":
+--
+-- It would be short-sighted (“myopic”) for stakeholders to directly use the
+-- reward splitting formulas from Section 6.5. They should instead take the
+-- long-term (“non-myopic”) view. To this end, the system will calculate and
+-- display the “non-myopic” rewards that pool leaders and pool members can
+-- expect, thus supporting stakeholders in their decision whether to create a
+-- pool and to which pool to delegate their stake.
+--
+-- The idea is to first rank all pools by “desirability”, to then assume that
+-- the k most desirable pools will eventually be saturated, whereas all other
+-- pools will lose all their members, then to finally base all reward
+-- calculations on these assumptions.
+--
+-- (Kant et al, 2019)
+
+-- | How desirable a pool would be if it were saturated.
+--
+-- (5.6.1 Pool Desirability and Ranking)
+desirability
+    :: RankingConstants
+    -> Pool
+    -> Double
+desirability env (Pool (RelativeStake leaderStake') (Lovelace c) (Margin m) (Bounded p))
+    | saturatedPoolRewards <= c = 0
+    | otherwise                 = (saturatedPoolRewards - c) * (1 - m)
+  where
+    -- | Total rewards for a pool if it were saturated.
+    saturatedPoolRewards :: Double
+    saturatedPoolRewards =
+        let
+            a0 = unBounded $ leaderStakeInfluence env
```

Comment:
I am still using the `env` variable name after renaming the type. Should change.